app/models/task.rb

```diff
@@ -36,4 +36,8 @@ def archived?
   def days_left
     (self.due_date - Date.today).to_i if self.due_date
   end
+
+  def overdue?
+    true unless days_left.nil? || days_left >= 0 || archived?
```

Comment:
I've been thinking about this line (great idea for a new method, by the way!), and I'm wondering if it might be more clear if it was:
```ruby
!archived && days_left.present? && days_left < 0
```
What do you think?